client/ayon_ftrack/ftrack_addon.py

```diff
@@ -178,6 +191,72 @@ def stop_timer(self):
         if self._tray_wrapper:
             self._tray_wrapper.stop_timer_manager()
 
+    def ensure_is_process_ready(self, context):
+        """Ensure addon is ready for process.
+
+        Args:
+            context (ProcessContext): Process context.
+
+        """
+        # Safe to support older ayon-core without 'ProcessPreparationError'
+        from ayon_core.addon import ProcessPreparationError
+        from ayon_ftrack.common import is_ftrack_enabled_in_settings
+
+        # Do not continue if Ftrack is not enabled in settings
+        if context.project_name:
+            settings = get_project_settings(context.project_name)
+        else:
+            settings = get_studio_settings()
+
+        if not is_ftrack_enabled_in_settings(settings):
+            return
+
+        # Not sure if this should crash or silently continue?
+        server_url = self.get_ftrack_url()
+        if not server_url:
+            return
+
+        username = os.getenv("FTRACK_API_USER")
+        api_key = os.getenv("FTRACK_API_KEY")
+
+        if (
+            username and api_key
+            and check_credentials(username, api_key, server_url)
+        ):
+            self.set_credentials_to_env(username, api_key)
+            return
+
+        username, api_key = self.get_credentials()
+        if (
+            username and api_key
+            and check_credentials(username, api_key, server_url)
+        ):
+            self.set_credentials_to_env(username, api_key)
+            return
+
+        if context.headless:
+            raise ProcessPreparationError(
+                "Ftrack credentials are not set. Cannot handle the situation"
+                " in headless mode."
```

Comment:
This needs some artist friendly message. Marking @BigRoy and @mkolar for suggestions.